scripts/build/builders/imx.py

```diff
@@ -74,30 +75,58 @@ def __init__(self,
         self.release = release
         self.app = app
 
-    def GnBuildEnv(self):
-        return {
-            'PKG_CONFIG_PATH': self.SysRootPath('IMX_SDK_ROOT') + '/sysroots/cortexa53-crypto-poky-linux/lib/aarch64-linux-gnu/pkgconfig',
-        }
-
     def GnBuildArgs(self):
+        entries = os.listdir(self.SysRootPath('IMX_SDK_ROOT'))
+        for entry in entries:
+            if re.match(r'^environment-setup-', entry):
+                env_setup_script = entry
+                break
+
+        with open(os.path.join(self.SysRootPath('IMX_SDK_ROOT'), env_setup_script), 'r') as env_setup_script_fd:
+            lines = env_setup_script_fd.readlines()
+            for line in lines:
+                line = line.strip('\n')
+                if re.match(r'export SDKTARGETSYSROOT=', line):
+                    sdk_target_sysroot = line[len(r'export SDKTARGETSYSROOT='):]
+                if re.match(r'export CC=', line):
+                    # remove the quotation marks, replace a shell env
+                    cc = line[len(r'export CC=')+1: -1]
+                    cc = cc.replace('$SDKTARGETSYSROOT', sdk_target_sysroot)
```

Comment:
Is it guaranteed that the lines for CC and CXX are after SDKTARGETSYSROOT ?

If not, the replace should be as a final step

Reply:
Tried in bash/zsh, the CC and CXX must be after SDKTARGETSYSROOT, or the variable SDKTARGETSYSROOT used in their value assigment cannot be correctly recognized.